index.bs

```diff
@@ -1462,3 +1484,231 @@ The [=remote end steps=] with |command parameters| are:
+    production with the <code>context</code> field set to |context id|, the
+    <code>parent</code> field set to |parent id|, and the <code>url</code> field
+    set to |url|.
+
+ 1. Return |context info|.
+
+</div>
+
+### Commands ### {#module-browsingContext-commands}
+
+#### list ####  {#command-browsingContext-list}
+
+The <dfn export for=commands>browsingContext.list</dfn> command returns a list
+of all browsing contexts that are children of the given context, or all
+top-level contexts when no parent is provided.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      BrowsingContextListCommand = {
+        method: "browsingContext.list",
+        params: BrowsingContextListParameters
+      }
+
+      BrowsingContextListParameters = {
+        ?parent: BrowsingContext,
+      }
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        BrowsingContextListResult = {
+          contexts: [*BrowsingContextInfo]
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.list">
+The [=remote end steps=] with |command parameters| are:
+
+  1. Let the |parent id| be the value of the <code>parent</code> field of
+       |command parameters| if present, or null otherwise.
+
+  1. Let |parent| be the result of [=trying=] to [=get a browsing context=]
+     given |parent id|.
+
+  1. If |parent| is null, let |contexts| be a list containing all [=top-level
+     browsing contexts=]. Otherwise let |contexts| be a list containing all
+     [=/browsing contexts=] which are [=child browsing contexts=] of |parent|.
+
+  1. Let |context info| be a list.
+
+  1. For each |context| of |context|, append the result of [=get the browsing
+     context info=] given |context| to |context info|.
+
+  1. Let |body| be a map matching the <code>BrowsingContextListResult</code>
+     production, with the <code>contexts</code> field set to |context info|.
+
+  1. Return [=success=] with data |body|.
+
+Issue: Should this contain other data like the current URL for the context.
+
+</div>
+
+### Events ### {#module-contexts-events}
+
+#### contextCreated #### {#event-browsingContext-contextCreated}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextContextCreatedEvent = {
+         method: "browsingContext.contextCreated",
+         params: BrowsingContextInfo
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for browsingContext.contextCreated">
+When the [=create a new browsing context=] algorithm is invoked, after the
+[=active document=] of the browsing context is set, run the following steps:
+
+ 1. Let |context| be the newly created browsing context.
+
+ 1. Let |related contexts| be a set containing the [=parent browsing context=]
+    of |context|, if that is not null, or an empty set otherwise.
```

Comment:
Is it possible for the parent to be null? My understanding was that iframes, for example, get a browsing context created when inserted into the DOM. How does one create an orphaned browsing context, and is that surfaced in other protocols, if at all?

I came across https://chromedevtools.github.io/devtools-protocol/tot/Page/#event-frameAttached, but I don't know if that's the only related event in CDP. @sadym-chromium do you know?

Reply:
Oh right, obviously, I was too zoomed in on observing a specific page.